k8s/pkg/apis/longhorn/v1beta1/backup.go

```diff
@@ -11,42 +11,62 @@ const (
 	BackupStateUnknown    = BackupState("Unknown")
 )
 
+// SnapshotBackupSpec defines the desired state of the Longhorn backup
 type SnapshotBackupSpec struct {
+	// The time to request run sync the remote backup.
 	// +optional
 	SyncRequestedAt metav1.Time `json:"syncRequestedAt"`
+	// The snapshot name.
 	// +optional
 	SnapshotName string `json:"snapshotName"`
+	// The labels of snapshot backup.
 	// +optional
 	Labels map[string]string `json:"labels"`
 }
 
+// SnapshotBackupStatus defines the observed state of the Longhorn backup
 type SnapshotBackupStatus struct {
+	// The owner ID which is responsible to reconcile this backup CR.
 	// +optional
 	OwnerID string `json:"ownerID"`
+	// The backup creation state.
+	// Can be "", InProgress, Completed, Error, Unknown.
```

Comment:
Add quotas for all enums: `Can be "", "InProgress", "Completed", "Error", "Unknown".`

Reply:
You mean add validation for the enum of status fields as well?

I tried to add validation for all the spec fields, but for status fields, it can only be updated by controller. Therefore, the end user should be unable to change the status field by kubectl.

If we would like to make all the status fields add enum validation, then we need to run full E2E test to double verify it as well.

Reply:
No. I just talked about the description here.